### ISSUE TITLE
Change the name of `apiFetch` of `@wordpress/data-controls` as it seems to be the same with `apiFetch` of `@wordpress/apiFetch`

### ISSUE THREAD

**Issue by hrrsppzgl** [WordPress/gutenberg#22252]
**Is your feature request related to a problem? Please describe.**
`import apiFetch from '@wordpress/api-fetch'`  vs `import { apiFetch } from '@wordpress/data-controls'` is somehow confusing as `apiFetch` seems to be (from naming) the same but the first is actually a wrapper of fetch() (that's cool) while the second is an action generator that eventually uses the first `apiFetch` so we can handle async calls in stores! Even if someone reads the documentation for both `@wordpress/data-controls` (https://github.com/WordPress/gutenberg/tree/master/packages/data-controls#apiFetch) and `@wordpress/apiFetch` (https://github.com/WordPress/gutenberg/tree/master/packages/api-fetch#options) it is very counter intuitive the first time.

**Describe the solution you'd like**
Maybe using `apiFetchGenerator` instead of `apiFetch` for `@wordpress/data-controls`would be more suitable? Can we alias and deprecate the old name so that we don't introduce a breaking change?

**Describe alternatives you've considered**
As an alternative, we can update our documentation for both packages warning the user that this two functions are actually very different

**Comment by chrisvanpatten** [comment]
I actually have read the data-controls docs several times and I don't think I picked up on the distinction either. I agree, and if there are backcompat concerns around changing the name a documentation change would be great.

**Comment by jordesign** [comment]
Hey @cpapazoglou  - I'm just going back through some older issues - wanted to see if this one is still relevant?

**Comment by cpapazoglou** [comment]
> Hey @cpapazoglou - I'm just going back through some older issues - wanted to see if this one is still relevant?

Yeah, the confusion still stands. Although, I haven't used any of them since then 😅

**Comment by talldan** [comment]
I'm going to close the issue. There's no action in years, so perhaps it's not worth it. It also looks like data-controls is not used internally now.

Not sure if it's a candidate for a complete deprecation as a package, it might still be used by third parties.
